src/components/LoginForm.jsx

```diff
@@ -0,0 +1,193 @@
+  }
+
+  handleRememberMeCheckBox = (e) => {
+    this.setState({ [e.target.id]: e.target.checked });
+  }
+
+  /**
+   * Save token to storage and redirect user
+   */
+  LoginUser = (rememberToken, token) => {
+    window.localStorage.setItem('token', token);
+    window.localStorage.setItem('rememberToken', rememberToken);
+    const { history } = this.props;
+    history.push('/dashboard');
+  }
+
+  processLogin = async () => {
+    const { dispatch } = this.props;
+    const { email, password, rememberLogin } = this.state;
+    await dispatch(loginUserRequest({ email, password }, rememberLogin));
+    const {
+      auth: {
+        status, message, rememberToken, token
+      }
+    } = this.props;
+    if (status === 200) {
+      toast(<ToastMessage message={`${message}. Redirecting...`} />, {
+        type: 'success',
+        closeButton: false,
+        hideProgressBar: true,
+        onClose: () => this.LoginUser(rememberToken, token),
+        autoClose: 0,
+      });
+    } else if (status === 400) {
+      toast(<ToastMessage message={message} />, {
+        type: 'error',
+        closeButton: false,
+        hideProgressBar: true,
+        autoClose: 5000,
+      });
+    } else {
+      toast(<ToastMessage message='Something very strange happened' />, {
+        type: 'error',
+        closeButton: false,
+        hideProgressBar: true,
+        autoClose: 5000,
+      });
+    }
+  }
+
+  checkEmailInput = (e) => {
+    const validEmailRegex = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;
+    if (!validEmailRegex.test(e.target.value)) {
+      this.setState({
+        emailError: 'Email is invalid',
+        isEmailValid: false,
+      });
+    } else {
+      this.setState({
+        emailError: '',
+        isEmailValid: true,
+      });
+    }
+  }
+
+  checkPasswordInput = (e) => {
+    const validPasswordRegex = /^(?=.*[a-zA-Z])(?=.*[0-9]){8}/;
+    if (!validPasswordRegex.test(e.target.value)) {
+      this.setState({
+        passwordError: 'Should be alphanumeric and atleast 8 characters long',
+        isPasswordValid: false,
+      });
+    } else {
+      this.setState({
+        passwordError: '',
+        isPasswordValid: true,
+      });
+    }
+  }
+
+  resetInput = (e) => {
+    if (e.target.id === 'email') {
+      this.setState({ emailError: '', isEmailValid: false });
+    }
+    if (e.target.id === 'password') {
+      this.setState({ passwordError: '', isPasswordValid: false });
+    }
+  }
+
+  render() {
+    const { auth: { isLoading } } = this.props;
+    const {
+      email, password, emailError, passwordError, isEmailValid, isPasswordValid
+    } = this.state;
+    const spinner = isLoading ? (<Spinner style={{ width: '1.5rem', height: '1.5rem' }} />) : null;
+    const disabled = !(isEmailValid && isPasswordValid) || isLoading;
+    return (
+      <div className='login-form'>
+        <h4 className='login-form-header-text'>Welcome back</h4>
+        <InputGroup className='login-form-input'>
```

Comment:
Can the input fields be made reusable also? Like define it once and then call it wherever it's needed with the props

Reply:
@Chiazokam I used the default `<Input />` `reactstrap` component as is so I don't think this needs to be modified further